assets/javascripts/discourse/components/chat-message-move-to-channel-modal-inner.js

```diff
@@ -0,0 +1,55 @@
+import Component from "@ember/component";
+import discourseComputed from "discourse-common/utils/decorators";
+import { isBlank } from "@ember/utils";
+import { action, computed } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+import { inject as service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default class MoveToChannelModalInner extends Component {
+  @service chat;
+  @service router;
+  tagName = "";
+  sourceChannel = null;
+  destinationChannelId = null;
+  selectedMessageIds = null;
+
+  @computed("destinationChannelId")
+  get disableMoveButton() {
+    return isBlank(this.destinationChannelId);
+  }
+
+  @discourseComputed("chat.publicChannels.[]")
+  availableChannels(publicChannels) {
+    return publicChannels.reject((chan) => chan.id === this.sourceChannel.id);
+  }
+
+  get selectedMessageCount() {
+    return this.selectedMessageIds.length;
+  }
+
+  @action
+  moveMessages() {
+    return ajax(
```

Comment:
Just curious what happens if I call the same endpoint with same params multiple times?

Reply:
Ok so this is what I thought. This is kinda unexpected for a PUT as it should more or less be idempotent. I don't want to go into a lengthy REST debate, but if you have an idea of something else than PUT here, that might be good

Reply:
I mean, I think it fits PUT as we are updating stuff. We don't exactly strictly adhere to REST in most places. I don't really care though happy to change to POST if you want.

Reply:
Also it is idempotent, you can't move the same messages twice. I intentionally throw an error too, it's not just an unhandled error: 

https://github.com/discourse/discourse-chat/blob/1d1707fd11fcf84d451f2fe80745b178b076c6d9/lib/message_mover.rb#L36-L38

Reply:
I just think maybe it should just not error if these message ids are already associated with this channel. But that will make code more complicated, I will live with this.